flytekit/core/local_cache.py

```diff
@@ -14,13 +14,13 @@
 
 def _recursive_hash_placement(literal: Literal) -> Literal:
     if literal.collection is not None:
-        literals = [_recursive_hash_placement(literal) for literal in literal.collection.literals]
-        return Literal(collection=LiteralCollection(literals=literals))
+        literals = [_recursive_hash_placement(lit) for lit in literal.collection.literals]
+        return Literal(collection=LiteralCollection(literals=literals), hash=literal.hash)
     elif literal.map is not None:
         literal_map = {}
-        for key, literal in literal.map.literals.items():
-            literal_map[key] = _recursive_hash_placement(literal)
-        return Literal(map=LiteralMap(literal_map))
+        for key, literal_value in literal.map.literals.items():
+            literal_map[key] = _recursive_hash_placement(literal_value)
+        return Literal(map=LiteralMap(literal_map), hash=literal.hash)
```

Comment:
now that I'm looking at this again, we should simplify this and apply the same idea that we use in the case of the remote cache, i.e. if the hash in the literal we don't have to recurse and can simply return an empty literal containing the hash.

Reply:
yeah, that is a good idea. I changed the loop structure to incorporate this idea, let's see what the tests say.